### ISSUE TITLE
 [clang-tidy][false negative] bugprone-optional-value-conversion not detected inside std::make_unique

### ISSUE THREAD

**Issue by hexagonrecursion** [llvm/llvm-project#110964]
[godbolt](https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,selection:(endColumn:2,endLineNumber:10,positionColumn:2,positionLineNumber:10,selectionStartColumn:2,selectionStartLineNumber:10,startColumn:2,startLineNumber:10),source:'%23include+%3Cmemory%3E%0A%23include+%3Coptional%3E%0A%0Aclass+C%7B%0Apublic:%0A++++C(std::optional%3Cint%3E)+%7B%7D%0A%7D%3B%0A%0Aint+main()%0A%7B%0A++std::optional%3Cint%3E+opt%3B%0A%0A++std::make_unique%3CC%3E(opt.value())%3B%0A%7D'),l:'5',n:'1',o:'C%2B%2B+source+%231',t:'0'),(h:compiler,i:(compiler:clang_trunk,filters:(b:'0',binary:'1',binaryObject:'1',commentOnly:'0',debugCalls:'1',demangle:'0',directives:'0',execute:'1',intel:'0',libraryCode:'0',trim:'1',verboseDemangling:'0'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,libs:!(),options:'',overrides:!(),selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1),l:'5',n:'0',o:'+x86-64+clang+(trunk)+(Editor+%231)',t:'0')),k:39.1644908616188,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:tool,i:(args:'--checks%3D!'-*,bugprone-optional-value-conversion!'',argsPanelShown:'0',compilerName:'x86-64+clang+(trunk)',editorid:1,fontScale:14,fontUsePx:'0',j:1,monacoEditorHasBeenAutoOpened:'1',monacoEditorOpen:'1',monacoStdin:'1',stdin:'',stdinPanelShown:'1',toolId:clangtidytrunk,treeid:0,wrap:'1'),l:'5',n:'0',o:'clang-tidy+(trunk)+x86-64+clang+(trunk)+(Editor+%231,+Compiler+%231)',t:'0')),k:60.835509138381205,l:'4',n:'0',o:'',s:0,t:'0')),l:'2',n:'0',o:'',t:'0')),version:4)

```c++
#include <memory>
#include <optional>

class C{
public:
    C(std::optional<int>) {}
};

int main()
{
  std::optional<int> opt;

  std::make_unique<C>(opt.value()); // should warn here, but does not
}
```

**Comment by HerrCai0907** [comment]
It depends on overload decision. Since clang-tidy will not involve Sema. I think it cannot be supported.
